fix: eventBus error after dispose

adrums86 opened the pull request:
fix an issue where the eventBus reference is destroyed before a `keystatuseschange` event is fired after a `despose()` call on the player.

misteroneill commented on src/eme.js:
```diff
@@ -116,12 +116,7 @@ export const makeNewRequest = (player, requestOptions) => {
     });
 
     player.on('dispose', () => {
-      keySession.close().then(() => {
-        eventBus.trigger({
-          type: 'keysessionclosed',
-          keySession
-        });
-      }).catch((error) => {
```
Since we're doing this on `dispose`, removing the `then()` makes the most sense. Since the `keySession.close()` is async, we will have disposed the tech (i.e. `eventBus`) by the time this promise resolves.

Future enhancement might be to delay the `dispose` until the key session is closed, but I'm not sure that would be useful or necessary.

adrums86 replied:
Do we want to trigger the `keysessionclosed` event anyway outside of the `then` block?

dzianis-dashkevich replied:
I am not sure about this change, maybe it will be safer to just: replace all occurrences of the `eventBus.trigger(...args)` to the `trigger(eventBus, ...args)`

Where `trigger` is

```js

function trigger(eventBus, ...args) {
  if (eventBus.isDisposed()) {
     return;
  }

  eventBus.trigger(...args);
}

```

misteroneill commented on src/eme.js:
```diff
@@ -210,6 +210,13 @@ export const makeNewRequest = (player, requestOptions) => {
+            // player has been disposed.
+            if (player.isDisposed()) {
+              return;
+            }
```
Added this one just in case.

misteroneill commented on src/eme.js:
```diff
@@ -168,6 +163,11 @@ export const makeNewRequest = (player, requestOptions) => {
+        // Protect from race conditions causing the player to be disposed.
+        if (player.isDisposed()) {
+          return;
+        }
```
Using `isDisposed()` is a somewhat cleaner way to check if we can trigger events.

dzianis-dashkevich commented:
I assume, we should update all occurrences across the plugin

adrums86 commented:
Fixed any other unsafe calls to `eventBus.trigger` and fixed remaining tests. Also update the lint standard to allow for use of the spread operator for objects.
